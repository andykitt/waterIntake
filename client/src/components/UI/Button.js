import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Btn = styled.button`
  height: 2.5rem;
  padding: 0.4rem 1rem;
  color: white;
  font-size: 1rem;
  margin: 0.3rem;
  border-radius: 0.2rem;
  border: none;
  background: ${props =>
    props.success
      ? '#23d160'
      : props.danger
      ? '#f22613'
      : props.primary
      ? '#209cee'
      : 'white'};

  &:focus {
    outline: 0;
    border: 0;
  }

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }

  @media (max-width: 768px) {
    height: 1.8rem;
    font-size: 0.6rem;
  }
`;

const Button = props => {
  return (
    <Btn
      width={props.width}
      danger={props.danger}
      primary={props.primary}
      success={props.success}
      type={props.type}
      onClick={props.onClick}
      disabled={props.disabled}
    >
      {props.children}
    </Btn>
  );
};

Button.propTypes = {
  width: PropTypes.string,
  danger: PropTypes.bool,
  primary: PropTypes.bool,
  success: PropTypes.bool,
  type: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};

export default Button;
