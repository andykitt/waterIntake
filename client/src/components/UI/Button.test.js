import React from 'react';
import Button from './Button';
import { render } from 'react-testing-library';

test('renders success with color #23d160', () => {
  const { container } = render(<Button success />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders danger with color #f22613', () => {
  const { container } = render(<Button danger />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders primary with color #209cee', () => {
  const { container } = render(<Button primary />);
  expect(container.firstChild).toMatchSnapshot();
});

test('renders primary with color white', () => {
  const { container } = render(<Button />);
  expect(container.firstChild).toMatchSnapshot();
});
