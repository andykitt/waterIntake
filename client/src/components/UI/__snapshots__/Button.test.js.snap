// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders danger with color #f22613 1`] = `
<button
  class="sc-bdVaJa fzjQEY"
  data-testid="button"
/>
`;

exports[`renders primary with color #209cee 1`] = `
<button
  class="sc-bdVaJa kZKjYZ"
  data-testid="button"
/>
`;

exports[`renders primary with color white 1`] = `
<button
  class="sc-bdVaJa bgDRfj"
  data-testid="button"
/>
`;

exports[`renders success 1`] = `
<button
  class="sc-bdVaJa ghaKnc"
  data-testid="button"
/>
`;

exports[`renders success with color #23d160 1`] = `
<button
  class="sc-bdVaJa ghaKnc"
  data-testid="button"
/>
`;
